client-app/shared/catalog/components/product-card-grid.vue

```diff
@@ -74,4 +80,16 @@ const props = defineProps({
 });
 
 const link = computed<RouteLocationRaw>(() => getProductRoute(props.product));
+
+const sale = computed(() => {
+  const {
+    list: { amount: listPrice },
+    sale: { amount: salePrice },
+  } = props.product.price;
```

Comment:
Please, fix `Property 'N' does not exist on type 'PriceType | undefined'` error message, it's needed to check that price is passed or explicitly convert to PriceType.